File: noxClassificationTest.py
import tfunHDDC as tfun
import numpy as np
import NOxBenchmark as NOx
from sklearn import metrics as met
import time

if __name__ == '__main__':
    models = ['akjbkqkdk', 'akjbqkdk', 'akbkqkdk', 'akbqkdk', 'abkqkdk', 'abqkdk']
    data = NOx.fitNOxBenchmark()
    training = np.arange(0,50)
    test = np.arange(50,100)
    known = data['labels'][training]
    labels = data['labels']

    results = {}
    for i in models:
        start = time.time()
        res = tfun.tfunHDDC(data['data'][training], min_individuals=4, model=i, K=2, threshold=[0.05, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6], nb_rep=20, init='kmeans', known = known, mc_cores=16)
        results[i] = [np.sum(np.diag(met.confusion_matrix(res.cl, known)))/len(known)]
        results[i].append(met.confusion_matrix(res.cl, known))
        
        results[i].append(tfun._T_hddc_ari(known, res.cl))
        results[i].append(f"Time taken: {time.time() - start} seconds")

        pred = res.predict(data['data'][test])
        results[i].append(np.sum(np.diag(met.confusion_matrix(pred['class'], labels[test])))/len(labels[test]))
        results[i].append(met.confusion_matrix(pred['class'], labels[test]))
        results[i].append(tfun._T_hddc_ari(labels[test], pred['class']))
        
    for i in results.values():
        print(i)

File: functionTests/miscTests/noxClassificationTest.R
library(TFunHDDC)

models = c("akjbkqkdk", 'akjbqkdk', 'akbkqkdk', 'akbqkdk', 'abkqkdk', 'abqkdk')
data = fitNOxBenchmark()
training = c(1:50)
test = c(51:100)
clm = data$groupd
known = clm[training]

results=list()
for(i in models){
  print(i)
  start = Sys.time()
  res = tfunHDDC(data$fd[training], K=2, model=i, known = known, threshold = c(0.05, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6), init = 'kmeans', nb.rep=20, verbose=FALSE) 
  temp1 = table(res$class, clm[training])
  temp2 = TFunHDDC:::.T_hddc_ari(clm[training], res$class)
  temp3 = Sys.time() - start
  pred = TFunHDDC::predict.tfunHDDC(res, data$fd[test])$class
  temp4 = table(pred, clm[test])
  temp5 = TFunHDDC:::.T_hddc_ari(pred, clm[test])
  print(temp1)
  print(temp2)
  print(temp3)
  print(temp4)
  print(temp5)
  print('\n')
}